src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/RoboticsAddressMapper.java

```diff
@@ -40,4 +48,73 @@ public RoboticsAddresses toRoboticsAddresses(List<ContactInformation> contactInf
             ? null
             : toRoboticsAddresses(fromContactInformation(contactInformation.get(0)));
     }
+
+    private Address tryToRollOverAddressLines(Address address) {
+        String[] addressLines = new String[]{
+            address.getAddressLine1(), address.getAddressLine2(), address.getAddressLine3()};
+
+        if (addressLines[0].length() > 35) {
+            tryToRollOverAddressLine(addressLines, 0);
+        }
+
+        if (addressLines[1] != null && addressLines[1].length() > 35) {
+            tryToRollOverAddressLine(addressLines, 1);
+        }
+
+        if (Arrays.stream(addressLines).filter(Objects::nonNull).anyMatch(addressLine -> addressLine.length() > 35)) {
+            throw new AddressLineExceedsLengthLimitException();
+        }
+
+        return Address.builder()
+            .addressLine1(addressLines[0])
+            .addressLine2(addressLines[1])
+            .addressLine3(addressLines[2])
+            .postTown(address.getPostTown())
+            .county(address.getCounty())
+            .country(address.getCountry())
+            .postCode(address.getPostCode())
+            .build();
+    }
+
+    private void tryToRollOverAddressLine(String[] addressLines, int addressLineIndex) {
+        String[] addressSplit = addressLines[addressLineIndex].split(", ");
+
+        if (addressSplit.length == 1) {
+            throw new AddressLineExceedsLengthLimitException();
+        }
+
+        addressLines[addressLineIndex] = String.join(
+            ", ",
+            ArrayUtils.subarray(addressSplit, 0, addressSplit.length - 1)
+        );
+
+        if (addressLines[addressLineIndex + 1] == null) {
```

Comment:
Could we also try to make use of collections (ArrayList) instead of manipulating arrays and indexes?

Reply:
As a bonus, one could re-write this recursively ;-)